OSSMDOC-550: Document new WasmPlugin API. (#45111)
openshift/openshift-docs: modules/ossm-extensions-wasmplugin-format.adoc

////
This module included in the following assemblies:
*service_mesh_/v2x/ossm-extensions.adoc
////
:_content-type: REFERENCE
[id="ossm-extensions-wasmplugin-format_{context}"]
= `WasmPlugin` container format

> Istio supports Open Container Initiative (OCI) images in its Wasm Plugin mechanism. You can distribute your Wasm Plugins in the form of a container image, and you can use `spec.url` to refer to a container registry location.  For example, `quay.io/my-username/my-plugin:latest`.

Reviewer: API object reference and minor suggestion:
```suggestion
Istio supports Open Container Initiative (OCI) images in its Wasm Plugin mechanism. You can distribute your Wasm Plugins as a container image, and you can use the `spec.url` field to refer to a container registry location.  For example, `quay.io/my-username/my-plugin:latest`.
```

Because each execution environment (runtime) for a WASM module can have runtime-specific configuration parameters, a WASM image can be composed of two layers:

> * *plugin.wasm* (Required) - Content layer. This layer consists of a `.wasm` binary containing the bytecode of your WebAssembly module, to be loaded by the runtime.  This file must be named `plugin.wasm`.

Reviewer: Minor nit for active voice:
```suggestion
* *plugin.wasm* (Required) - Content layer. This layer consists of a `.wasm` binary containing the bytecode of your WebAssembly module, to be loaded by the runtime.  You must name this file `plugin.wasm`.
```

> * *runtime-config.json* (Optional) - Configuration layer. This layer consists of a JSON-formatted string that describes metadata describing the module for the target runtime. The config layer may also contain additional data, depending on the target runtime. For example, the config for a WASM Envoy Filter contains root_ids available on the filter.

Reviewer: ISG nit:
```suggestion
* *runtime-config.json* (Optional) - Configuration layer. This layer consists of a JSON-formatted string that describes metadata about the module for the target runtime. The config layer might also contain additional data, depending on the target runtime. For example, the config for a WASM Envoy Filter contains root_ids available on the filter.
```
